lib/cocoapods-core/specification/json.rb

```diff
@@ -22,7 +22,6 @@ def to_pretty_json(*a)
       #
       def to_hash
         hash = attributes_hash.dup
-        hash.delete('platforms')
```

Comment:
so if we keep this or even move it into the conditional it appears it works fine. _However_, if you pre-load an existing JSON podspec and we keep this line here we actually end up shifting the hash causing a diff. This is why this line should be deleted entirely.

I will be adding an additional test for this.

Reply:
Test added